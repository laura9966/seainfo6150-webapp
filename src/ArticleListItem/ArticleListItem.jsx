import React from "react";
import { useState} from "react";
import PropTypes from "prop-types";
import styles from "./ArticleListItem.module.css";
import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton";
import ArticleImage from '../ArticleImage/ArticleImage.jsx';

const ArticleListItem = (props) => {
 const [showArticle, setShowArticle] = useState(false);
 const [buttonText, setButtonText] = useState("show more");
    const onClick =()=> {
            setShowArticle(!showArticle);
            if(!showArticle) {
              setButtonText("show less");
            } else {
              setButtonText("show more");
            }

    }
  return(
       <article className = {styles.box}>

            <div className={styles.articleImage}>
               <ArticleImage url={props.article.image._url} title={props.article.title}/>
             </div>

        <h2 className = {styles.header}>{props.article.title}</h2>
        {showArticle &&
    <div>
      <p className = {styles.summary}>{props.article.shortText}</p>
      <time className={styles.time} dateTime={props.article.timeStamp}>{props.article.displayDate}</time>
    </div>
    }
    <ArticleTextToggleButton onClick={onClick} buttonText={buttonText}/>
  </article>
  );
};

ArticleListItem.propTypes = {
  article: PropTypes.object.isRequired
};

export default ArticleListItem;