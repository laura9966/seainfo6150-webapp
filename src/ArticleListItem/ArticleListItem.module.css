.box {
    background-image: linear-gradient(to right, rgba(204, 0, 0, .5), #eee 20% ) ;
    border: 1px solid #c00;
    border-radius: 10px;
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, .3);
    font-family: Arial, Helvetica, sans-serif;
    margin: 10px;
    padding: 20px;
    display: flex;
    flex-direction: column;
    flex:1;
    justify-content: space-between;

}
.articleImage{
   width: 100%;
}

.header {
    color: #c00;
    text-decoration: underline;
    margin: 0 0 20px;
}

.summary {
    font-size: 12px;
    margin: 0 0 20px;
}

.time {
    font-weight: bold;
}


@media screen and (max-width: 600px) {
    .header {
        font-size: 15px;
    }
    .summary {
        font-size: 12px;
    }
    .time {
        font-size: 12px;
    }
}

@media screen and (min-width: 600px) {
    .header {
        font-size: 15px;
    }
    .summary {
        font-size: 12px;
    }
     .time {
        font-size: 12px;
    }
}

@media screen and (min-width: 1000px) {
    .header {
        font-size: 20px;
    }
    .summary {
        font-size: 16px;
    }
    .time {
        font-size: 16px;
    }
}