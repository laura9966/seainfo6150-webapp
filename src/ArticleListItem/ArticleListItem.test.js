import React from "react";
import ArticleListItem from "./ArticleListItem";
import { fireEvent } from "@testing-library/react";
describe("ArticleListItem tests", () => {
  it("renders correctly", () => {
    const article1 = {

            title : "A title",
            timeStamp: "2020-02-02",
            displayDate: "2020-02-03"

    }
    const { container } = render(<ArticleListItem article={article1}/>);
    expect(container).toMatchSnapshot();
  });
});