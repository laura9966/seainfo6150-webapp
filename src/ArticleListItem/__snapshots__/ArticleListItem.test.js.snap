// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <article
    class="box"
  >
    <div
      class="articleImage"
    >
      <img
        alt="The Statue of Liberty's torch heads to new museum"
        class="image"
        src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
      />
    </div>
    <h2
      class="header"
    >
      The Statue of Liberty's torch heads to new museum
    </h2>
    <button
      class="button"
    >
      show more
    </button>
  </article>
</div>
`;
